core/lib/dal/src/consensus_dal.rs

```diff
@@ -26,15 +32,17 @@ impl ConsensusDal<'_, '_> {
                 fake_key
             "#
         )
-        .fetch_optional(self.storage.conn())
+        .try_map(|row| {
+            row.genesis
+                .map(|genesis| {
+                    zksync_protobuf::serde::deserialize(genesis).decode_column("genesis")
+                })
+                .transpose()
+        })
+        .instrument("genesis")
```

Comment:
nit: why instrumenting a future, rather than transforming the result?

Reply:
More opportunities for the instrumentation logic, some of which are used already (e.g., logic related to slow requests and measuring request latency).